### PR TITLE
App/song matcher

### PR DESCRIPTION
**Pull request by SgtTepper**
Template - just for you to play with

### CODE REVIEW

**Comment by thebne** [comment]
@SgtTepper add shit to requirements.txt

File: client/src/apps/chord-recognizer/src/index.js [hunk 1]
```diff
@@ -25,13 +25,14 @@ export default function ChordRecognizer({chords}) {
         }), 4000)
       }
       // add new detections
-      if ((!prev || prev.id !== chords.id) && chords.detection.length) {
+      if ((!prev || prev.id !== chords.id) && chords.detection && chords.detection.length) {
```
Comment (thebne) [inline client/src/apps/chord-recognizer/src/index.js:28]: remove that, it shouldn't happen anymore

File: client/src/apps/chord-recognizer/src/index.js [hunk 2]
```diff
@@ -91,7 +92,7 @@ export default function ChordRecognizer({chords}) {
 				}}
 			>
 				{({circle, g}) => {
-					const [main, ...rest] = animation.detection
+					const [main, ...rest] = animation.detection.sort(function(a, b){ return a.length - b.length; });
```
Comment (thebne) [inline client/src/apps/chord-recognizer/src/index.js:95]: add this logic to `utils/chords.js` instead

File: client/src/apps/chord-recognizer/src/index.js (hunk 1 above)
Comment (SgtTepper) [reply client/src/apps/chord-recognizer/src/index.js:28]: still happening, please remove this when you fix it

File: client/src/apps/chord-recognizer/src/index.js [hunk 3]
```diff
@@ -115,9 +116,9 @@ export default function ChordRecognizer({chords}) {
 									dominantBaseline: "hanging",
 									textAnchor: "middle",
 								}} >
-									{secondary}
```
Comment (thebne) [inline client/src/apps/chord-recognizer/src/index.js:118]: ===

**Pull request merged**